virttest/qemu_devices/qcontainer.py

```diff
@@ -1061,16 +1061,22 @@ def pflash_handler(firmware_name, cmd):
                 first_image = images[0]
                 img_params = params.object_params(first_image)
                 img_params["backing_chain"] = "no"
-                img_info = json.loads(qemu_storage.QemuImg(img_params,
-                                                           current_data_dir,
-                                                           first_image).info(True, "json"))
-                img_path, img_name = os.path.split(img_info.get("filename"))
-
-                pflash_vars_name = "%s_%s_VARS.fd" % (self.vmname, img_name)
-                pflash_vars_path = os.path.join(img_path, pflash_vars_name)
-                # When image has backing files, treat it as a temporary image
-                if "backing-filename" in img_info:
-                    self.temporary_image_snapshots.add(pflash_vars_path)
+                img_obj = qemu_storage.QemuImg(img_params,
+                                               current_data_dir, first_image)
+                if img_obj.is_remote_image():
+                    pflash_vars_name = "%s_%s_VARS.fd" % \
+                        (self.vmname, params['image_backend'])
```

Comment:
If only `vmname` and `backend` have used, I am afraid some different guests will use the same fd file, since avocado always use `avocado-vt-vm1` as the vmname.

Reply:
For the network-storage guest, the image is relatively single. The case also wants to be simple, sorry, I didn't think of the scene you were worried about.

Reply:
If there have 2 different guests in a loop, when they run the same storage case, will use the same fd file.

Reply:
Thanks a lot. after discussion, will add the image version in the vmname.

Reply:
Yep, some fields need to be considered, blk/scsi, qcow2/raw, guest versions...

Reply:
`avocado-vt-vm1_rhel840-aarch64_iscsi_direct_qcow2_VARS.fd` 
Since it is a network_storage, `blk/scsi` is not involved. Use `image_backend` instead.
Update done.